src/test/java/core/basesyntax/RegistrationServiceTest.java

```
@@ -0,0 +1,185 @@
+        user.setAge(18);
+        boolean actual = registrationService.register(user).getAge() == MIN_AGE;
+        assertTrue(actual, "Your age is lower than " + MIN_AGE + ".");
+    }
+
+    @Test
+    public void register_duplicateLogin_notOk() {
+        User user1 = new User();
+        user1.setLogin("admin1");
+        user1.setPassword("qwert1234");
+        user1.setAge(31);
+        registrationService.register(user1);
+        user.setLogin("admin1");
+        RegistrationServiceException ex = assertThrows(
+                RegistrationServiceException.class,
+                () -> registrationService.register(user)
+        );
+        assertEquals("Your login " + user.getLogin() + " already exists.", ex.getMessage());
+    }
+
+    @Test
+    public void register_validUser_ok() {
+        User user1 = new User();
+        user1.setLogin("admin1");
+        user1.setPassword("qwert1234");
+        user1.setAge(23);
+        registrationService.register(user1);
+        boolean actual = storageDao.get(user1.getLogin()).equals(user1);
+        assertTrue(actual, "Your login has been successfully registered!");
+    }
+
+    @Test
+    public void register_emptyPassword_NotOk() {
+        user.setPassword("");
+        RegistrationServiceException ex = assertThrows(
+                RegistrationServiceException.class,
+                () -> registrationService.register(user)
+        );
+        assertEquals("Your password must not be less than "
+                + MIN_PASSWORD_LENGTH + " characters!", ex.getMessage());
+    }
+
+    @Test
+    public void register_emptyLogin_NotOk() {
+        user.setLogin("");
+        RegistrationServiceException ex = assertThrows(
+                RegistrationServiceException.class,
+                () -> registrationService.register(user)
+        );
+        assertEquals("Your login length should be at least "
+                + MIN_PASSWORD_LENGTH + " characters!", ex.getMessage());
```

The empty-login test expects an error message that references the password-length constant: `+ MIN_PASSWORD_LENGTH +` rather than the login-length constant. This violates the checklist guidance that exception messages should be field-specific and that constants should be used consistently: "Exception messages should be informative (see code example above)." and "Move all hardcoded values to [constant fields] and give them informative names." Update the expected message to use `MIN_LOGIN_LENGTH` and ensure wording matches the service's login-length error message.